### PR TITLE
Improve boolean flag documentation generation

### LINKED ISSUE #47: Improve documentation for boolean flags with `true` as default value
@thaJeztah @crazy-max Following up on our discussion in https://github.com/docker/cli/pull/4259#discussion_r1483128896 and in https://github.com/docker/cli-docs-tool/pull/19#issuecomment-1937006531.

We should add values in `Type` and `Default` columns for boolean flags that have `true` as default value.

I'll be opening a PR soon!

### PR DESCRIPTION
**Pull request by gmargaritis**
Resolves #47 

Updated the documentation generation logic to include values in Type and Default columns for boolean flags that have true as a default value. This change addresses the need for users to differentiate between boolean flags that require explicit negation.

### CODE REVIEW

**Comment by gmargaritis** [comment]
@thaJeztah Let me know if we want to also display the Type of boolean flags regardless of their default value

**Comment by crazy-max** [comment]
I think we want the same for yaml generation?

**Comment by thaJeztah** [comment]
@crazy-max I think we already unconditionally set the default value for YAML; at least looking at the fixture here, I see `"false"` as default; https://github.com/docker/cli-docs-tool/blob/f5886777bdc70a58b86fed02c32abae511935ada/fixtures/docker_buildx_build.yaml#L74-L78

**Comment by gmargaritis** [comment]
That's correct!

https://github.com/docker/cli-docs-tool/blob/f5886777bdc70a58b86fed02c32abae511935ada/clidocstool_yaml.go#L271-L290

File: clidocstool_md.go [hunk 1]
```diff
@@ -253,7 +253,7 @@ func mdCmdOutput(cmd *cobra.Command, old string) (string, error) {
 				} else if cd, ok := cmd.Annotations[annotation.CodeDelimiter]; ok {
 					defval = strings.ReplaceAll(defval, cd, "`")
 				}
-			} else if f.DefValue != "" && (f.Value.Type() != "bool" && f.DefValue != "true") && f.DefValue != "[]" {
+			} else if f.DefValue != "" && ((f.Value.Type() != "bool" && f.DefValue != "true") || (f.Value.Type() == "bool" && f.DefValue == "true")) && f.DefValue != "[]" {
```
Comment (thaJeztah) [inline clidocstool_md.go:256]: For a follow-up, we should probably look at splitting some of these up. Things start to become too complicated IMO (too many boolean conditions here, which makes it easy to introduce bugs).

**Review by thaJeztah — APPROVED** [review]
LGTM

we should probably look at some additional test-cases in a follow-up

**Comment by thaJeztah** [comment]
@crazy-max LGTY?

**Comment by crazy-max** [comment]
Can we have a test in https://github.com/docker/cli-docs-tool/blob/main/clidocstool_test.go#L38? Just a dummy new flag like `--detach` that will default to true and update fixtures as well in https://github.com/docker/cli-docs-tool/tree/main/fixtures

**Comment by gmargaritis** [comment]
@crazy-max Added the test case, we should be good to go!

**Pull request merged**